### PR TITLE
Factor-out cross-compilation targets, add Linux cross

### LINKED ISSUE #67: Support more Linux cross-compile archs.
In addition to mingw, it would be nice to support other Linux cross compile targets.  eg. arm and aarch64.

Ubuntu provides some toolchains:

https://packages.ubuntu.com/source/focal/gcc-10-cross

https://packages.ubuntu.com/source/impish/gcc-11-cross

Questions I see:

* How to select targets?

Mingw has eg. `MINGW=32`.  Does each arch. get its own env. variable?  Or one variable with a list of values?

One obvious idea would be to reuse the `CROSS_COMPILER_TARGET_ARCHS` variable and the Base target names.

### PR DESCRIPTION
**Pull request by minijackson**
~~As a Draft PR to see if it's going in the right direction, and because I can't seem to run GitHub actions on my fork?~~

This adds support for many Linux cross-compilation targets, and factor that with WINE and RTEMS cross-compilation.

Fixes #67

### CODE REVIEW

**Comment by minijackson** [comment]
Removing the draft status since the github actions don't seem to run on draft PRs

**Comment by minijackson** [comment]
@mdavidsaver This is what I intended to do, but it seems that GHA didn't trigger in my fork (e.g: commit [0bcb170](https://github.com/minijackson/ci-scripts/commit/0bcb170f9103e5a69342edabcc4fee2a2360b90e)). I'm quite new to GHA actions, so I'm not sure what's wrong.

**Comment by minijackson** [comment]
I will make a PR changing the epics-base ci workflow if and once this PR is merged.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04, ubuntu-18.04, ubuntu-16.04, windows-2019, windows-2016, macos-10.15]
```
Comment (ralphlange) [inline .github/workflows/build-and-test.yml:22]: windows-2016 is actually still supported on GitHub Actions.

**Comment by ralphlange** [comment]
@mdavidsaver: Please also have a look at this - looks like a reasonable approach to me.

File: README.md [hunk 2]
```diff
@@ -310,6 +310,18 @@ RTEMS 5 builds need to be switched to a newer ubuntu version
...
+
+Setting the `LINUX_CROSS_<name>` environment variable enables cross-compiling
+to Linux for the provided target. The value of the environment variable should
+be the compiler prefix to be used.
+
+For example, to cross-compile for linux-arm with hard floating points
+
+- ensure you have the `g++-arm-linux-gnueabihf` package installed, if your CI
+  is on Ubuntu or Debian
+- set the environment variable `LINUX_CROSS_arm=arm-linux-gnueabihf`
```
Comment (mdavidsaver) [inline README.md:323]: It would be helpful to have a list of supported/tested possibilities here, as well as in `.github/workflows/build-and-test.yml`.

**Review by mdavidsaver — CHANGES_REQUESTED** [review]
Testing this PR on https://github.com/mdavidsaver/epics-base with no changes to my configuration gives errors on most jobs.  eg.

https://github.com/mdavidsaver/epics-base/runs/7201281364?check_suite_focus=true#step:5:46

```
Configuring EPICS build system
  EPICS Base build system set up for optimized build with shared (DLL) linking
  WARNING: deprecated WINE environment variable was specified, please use CI_CROSS_TARGETS
  Traceback (most recent call last):
    File ".ci/cue.py", line 1408, in <module>
      main(sys.argv[1:])
    File ".ci/cue.py", line 1404, in main
      args.func(args)
    File ".ci/cue.py", line 1126, in prepare
      prepare_cross_compilation(cross_target_info)
    File ".ci/cue.py", line 856, in prepare_cross_compilation
      raise ValueError(
  ValueError: Unknown CI_CROSS_TARGETS WINE. Please see the ci-scripts README for available values.
```

With similar errors for the RTEMS builds.

A couple of thoughts.

* `handle_old_cross_variables()` should log `$CI_CROSS_TARGETS` after editing.
* `handle_old_cross_variables()` could give a better suggestion.  eg. `Replace RTEMS=X with CI_CROSS_TARGETS=Y`.
* If we're going to continue to support the old `$RTEMS` and `$WINE`, then they should be covered by the test matrix.
* Should we bother supporting `$RTEMS` and `$WINE`, or force a migration now?

**Pull request closed without merging**